### ISSUE TITLE
Merge authority

### ISSUE THREAD

**Issue by Azhrei** [RPTools/maptool#64]
I haven't worked on MT in a *LONG* time.  Should I have merge authority over my own stuff?  I started this issue to provoke discussion...

This would mean Craig merges stuff I submit and I merge stuff he submits.  This puts at least one other set of eyeballs on the stuff he and I do ourselves.  Or we simply allow anyone with merge authority to merge anything.

If we go with the one's-code-can-only-be-merged-by-someone-else approach, there should probably be another person added to the pool of those with merge authority, simply because I don't hang out here after enough.

Perhaps this isn't necessary.  I know in my own private repo where my brother and nephew also have write authority, we have agreed that only my brother can merge to **master** as the project is really his baby, not ours.  But we've also been kicking around the idea of merges being done by someone other than the creator, just as a double-check on making stupid mistakes. :)

**Comment by joshsziegler** [comment]
To answer that, you probably want to think about:

1. The code being correct (bug free) and/or
2. The code architecture/design being "right"
3. How long are you willing to wait for someone to review the PR/merge

If you are just trying to avoid bugs, I'm not really sure how many of those are being caught at the PR/merge level. If you are trying to force a discussion about how features are implemented, then I would say that forcing someone else to review it might do some good. 

Then again I'm mostly watching MT develop from the sidelines, so this is just my opinion. :)

**Comment by JamzTheMan** [comment]
I'm not sure we really need the bottle neck given that we all can review the code easy enough, although I suggest we get into the habit of creating branches and building releases from those after a period of review. And then only after a period of time merge that branch back to master which should keep the matter clean and ready to build at any given time.

**Comment by cwisniew** [comment]
I am with jamz on this one, we need no more bottle necks. I have created a 1.4.0-dev-branch (a 1.4.1-dev-branch already exists). From this day we will just merge into the appropriate branch and then once a release is built merge that back into the master. Sure some things may need to be changed that may have been caught by a code review, but lets operate with the optimistic view that going back and changing things will in the end add up to less time than the waiting for reviews.

**Comment by Jaggeroth** [comment]
@Az well Slack is turning out to be a good way to keep up to speed on what is going on :)

Since we now seem to have some continuous integration going on, should we perhaps spend some time working on some tests? I say this acknowledging that none of my work projects have adequate tests ;)

**Comment by Azhrei** [comment]
I suppose I ought to sign up there.  I had heard that their licensing was such that they own anything uploaded there and being as they're not "free software" folks, I had been ignoring them. :-/

Has anyone else seen http://nvie.com/posts/a-successful-git-branching-model/ ?  I've been using it for about 6 months and I really like it...

**Comment by JamzTheMan** [comment]
I didn't know that about slack and suppose we could use gitter? But I doubt we're talking about anything worth keeping IP. That and Craig already had to stare at slack for work so we get attention easier :)

**Comment by Jaggeroth** [comment]
I love slack.

I tried to do a google for slack and ip issues but couldn't find anything. Want to share any links Az?

**Comment by Azhrei** [comment]
Don't look at me!  I think it was Craig's idea...

**Comment by Jaggeroth** [comment]
I meant about there being any IP concerns with Slack.
